rules/falco_rules.yaml

```diff
@@ -956,6 +956,19 @@
   priority: ERROR
   tags: [filesystem, mitre_persistence]
 
+- rule: Directory traversal monitored file read
+  desc: >
+    Web applications can be vulnerable to directory traversal attacks that allow accessing files outside of the web app's root directory (e.g. Arbitrary File Read bugs).
+    System directories like /etc are typically accessed via absolute paths. Access patterns outside of this (here path traversal) can be regarded as suspicious.
+  condition: open_read and (etc_dir or fd.name contains ".ssh/" or fd.name contains "id_rsa") and fd.nameraw contains "../" and fd.nameraw glob *../*../* and not proc.pname in (shell_binaries)
```

Comment:
great the `etc_dir` macro. What you think of using `user_ssh_directory` macro instead of `fd.name contains ".ssh/"` to check the user ssh directory?
also I agree with you re the position 👍🏻

Reply:
Done, now also `directory_traversal` macro, we are slowly closing in 🙃 .

Suggesting to optimize `user_ssh_directory`? Running glob against `fd.name` across few rules may have perf impacts. For instance, Big Data technologies are best used via pushing down cheap filters like `startswith`, `contains`, `endswith` ... before adding more expensive expressions.